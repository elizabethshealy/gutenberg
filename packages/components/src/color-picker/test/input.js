import TestRenderer from 'react-test-renderer';
import { Input } from '../inputs';
import { DOWN, ENTER, SPACE, UP } from '@wordpress/keycodes';

describe( 'Input ', () => {
	describe( 'calls onChange prop', () => {
		test( 'onKeyDown event for keyDown = ENTER', () => {
			const onChange = jest.fn();
			const testInstance = TestRenderer.create(
				<Input
					label={ 'Color value in hexadecimal' }
					valueKey="hex"
					value={ '#fff' }
					onChange={ onChange }
				/>
			).root;
			testInstance.findByType( 'input' ).props.onKeyDown( { keyCode: ENTER } );
			expect( onChange ).toHaveBeenCalledTimes( 1 );
		} );

		test( 'onKeyDown event for keyDown = UP', () => {
			const onChange = jest.fn();
			const testInstance = TestRenderer.create(
				<Input
					label={ 'Color value in hexadecimal' }
					valueKey="hex"
					value={ '#fff' }
					onChange={ onChange }
				/>
			).root;
			testInstance.findByType( 'input' ).props.onKeyDown( { keyCode: UP } );
			expect( onChange ).toHaveBeenCalledTimes( 1 );
		} );

		test( 'onKeyDown event for keyDown = DOWN', () => {
			const onChange = jest.fn();
			const testInstance = TestRenderer.create(
				<Input
					label={ 'Color value in hexadecimal' }
					valueKey="hex"
					value={ '#fff' }
					onChange={ onChange }
				/>
			).root;
			testInstance.findByType( 'input' ).props.onKeyDown( { keyCode: DOWN } );
			expect( onChange ).toHaveBeenCalledTimes( 1 );
		} );

		test( 'onChange event for value.length > 4', () => {
			const onChange = jest.fn();
			const testInstance = TestRenderer.create(
				<Input
					label={ 'Color value in hexadecimal' }
					valueKey="hex"
					value={ '#fff' }
					onChange={ onChange }
				/>
			).root;
			testInstance.findByType( 'input' ).props.onChange( { target: { value: '#ffffff' } } );
			expect( onChange ).toHaveBeenCalledTimes( 1 );
		} );

		test( 'onBlur event', () => {
			const onChange = jest.fn();
			const testInstance = TestRenderer.create(
				<Input
					label={ 'Color value in hexadecimal' }
					valueKey="hex"
					value={ '#fff' }
					onChange={ onChange }
				/>
			).root;
			testInstance.findByType( 'input' ).props.onBlur();
			expect( onChange ).toHaveBeenCalledTimes( 1 );
		} );
	} );

	describe( 'does not call onChange prop', () => {
		test( 'onKeyDown event when keyDown is not ENTER, DOWN, or UP', () => {
			const onChange = jest.fn();
			const testInstance = TestRenderer.create(
				<Input
					label={ 'Color value in hexadecimal' }
					valueKey="hex"
					value={ '#fff' }
					onChange={ onChange }
				/>
			).root;
			testInstance.findByType( 'input' ).props.onKeyDown( { keyCode: SPACE } );
			expect( onChange ).not.toHaveBeenCalled();
		} );

		test( 'onChange when value.length <= 4', () => {
			const onChange = jest.fn();
			const testInstance = TestRenderer.create(
				<Input
					label={ 'Color value in hexadecimal' }
					valueKey="hex"
					value={ '#fff' }
					onChange={ onChange }
				/>
			).root;
			testInstance.findByType( 'input' ).props.onChange( { target: { value: '#fff' } } );
			expect( onChange ).not.toHaveBeenCalled();
		} );
	} );

	describe( 'is re-rendered', () => {
		test( 'onChange event for value.length <= 4', () => {
			const onChange = jest.fn();
			const testRenderer = TestRenderer.create(
				<Input
					label={ 'Color value in hexadecimal' }
					valueKey="hex"
					value={ '#aaa' }
					onChange={ onChange }
				/>
			);
			testRenderer.root.findByType( 'input' ).props.onChange( { target: { value: '#fff' } } );
			expect( testRenderer.toJSON() ).toMatchSnapshot();
		} );

		test( 'onChange event for value.length > 4', () => {
			const onChange = jest.fn();
			const testRenderer = TestRenderer.create(
				<Input
					label={ 'Color value in hexadecimal' }
					valueKey="hex"
					value={ '#aaa' }
					onChange={ onChange }
				/>
			);
			testRenderer.root.findByType( 'input' ).props.onChange( { target: { value: '#ffffff' } } );
			expect( testRenderer.toJSON() ).toMatchSnapshot();
		} );
	} );
} );
