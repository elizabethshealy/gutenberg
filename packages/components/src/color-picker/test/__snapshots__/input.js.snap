// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input  is re-rendered onChange event for value.length <= 4 1`] = `
<div
  className="components-base-control components-color-picker__inputs-field"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-text-control-7"
    >
      Color value in hexadecimal
    </label>
    <input
      className="components-text-control__input"
      id="inspector-text-control-7"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
      value="#fff"
    />
  </div>
</div>
`;

exports[`Input  is re-rendered onChange event for value.length > 4 1`] = `
<div
  className="components-base-control components-color-picker__inputs-field"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-text-control-8"
    >
      Color value in hexadecimal
    </label>
    <input
      className="components-text-control__input"
      id="inspector-text-control-8"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
      value="#ffffff"
    />
  </div>
</div>
`;
